app/Services/ImportResults.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\ImportMeta;
+use App\Models\Mismatch;
+
+class ImportResults
+{
+    public function generateCSV(string $path, int $import_id): void
+    {
+        $mismatches = Mismatch::get();
+
+        // Creating the output stream
+        $handle = fopen($path, 'w');
+
+        $import = ImportMeta::find($import_id);
+        $mismatches = Mismatch::whereBelongsTo($import)->get();
+
+        $mismatchValues = array_map(function ($mismatch) {
```

Comment:
I think instead of  mapping the whole result at once, and then iterating over the resulting array, we could also iterate over the mismatches directly and call `fputcsv()` for each row?

Reply:
I ended up making this a LazyCollection, where the mapping is only triggered by the `.each()` iteration which writes the mismatch to the csv